src/services/link/index.js

```diff
@@ -94,7 +94,7 @@ module.exports = function setup() {
 			}
 
 			// base link
-			if (data.role === 'student') {
+			if (data.role === 'student' || (data.role.name && data.role.name === 'student')) {
 				linkData.link = `${data.host || Configuration.get('HOST')}/registration/${data.schoolId}`;
```

Comment:
where is this data sent from? it should be fixed there because it might lead to other bugs

Reply:
In nuxt, data sent as ['student'], and it's used in files and test files. In legacy-client, it is sent as 'student' and is also used in many files. So changing the data from clients is complicated.

The logic in this line is combined with the '||' operator. After the operator, it checks if "data.role.name" exists, if so it checks its value equals "student".  So it doesn’t cause a bug.